### ISSUE TITLE
[Rancher v1.2.1] Error: Unable to request an upgrade on Rancher

### ISSUE THREAD

**Issue by bigfoot90** [cdrx/rancher-gitlab-deploy#6]
I'm having some trouble with `rancher:v1.2.1`
```
$ upgrade --rancher-url $RANCHER_URL --rancher-key $RANCHER_ACCESS_KEY --rancher-secret $RANCHER_SECRET_KEY --environment Default --stack DataWarehouse --service code --new-image $CONTAINER_RELEASE_IMAGE_STABLE
Upgrading datawarehouse/code in environment Default...
Error: Unable to request an upgrade on Rancher
ERROR: Build failed: exit code 1
```

I've followed API to the url `/v1/projects/1a5/services/1s1049` but in `actions` I see only `remove` action.
The `ACCESS_KEY` and `SECRET_KEY` has been created by administrator as `Account API Keys`, I am missing something?

**Comment by cdrx** [comment]
Thanks for the bug report.

Are you able to upgrade the service, using that account, from the Rancher UI? Did this work for you on 1.2.0?

**Comment by djskinner-sky** [comment]
I'm using 1.2.1 and haven't had a problem:

```
deploy_test:
  stage: deploy
  image: cdrx/rancher-gitlab-deploy
  script:
    - upgrade --environment Default --stack mystack --service app --new-image $CONTAINER_TEST_IMAGE
```

Can you actually perform the upgrade in the rancher API itself?

**Comment by bigfoot90** [comment]
HI, thanks all for responces.

This is my `.gitlab.ci.yml`:
```yml
deploy_stable:
  stage: deploy
  image: cdrx/rancher-gitlab-deploy
  script:
    - upgrade
        --environment Default
        --stack mystack
        --service myapp
        --new-image $IMAGE_RELEASE_STABLE
```
Wich results in
```
$ upgrade --environment Default --stack mystack --service myapp --new-image $IMAGE_RELEASE_STABLE
Upgrading datawarehouse/code in environment Default...
Error: Unable to request an upgrade on Rancher
ERROR: Build failed: exit code 1
```

The script calls the url `/v1/projects/1a5/services/1s1049?action=upgrade`, but this url responds with
```json
{
"id": "1s1049",
"type": "composeService",
"links": {
"self": "…/v1/projects/1a5/composeservices/1s1049",
"account": "…/v1/projects/1a5/composeservices/1s1049/account",
"consumedbyservices": "…/v1/projects/1a5/composeservices/1s1049/consumedbyservices",
"consumedservices": "…/v1/projects/1a5/composeservices/1s1049/consumedservices",
"instances": "…/v1/projects/1a5/composeservices/1s1049/instances",
"networkDrivers": "…/v1/projects/1a5/composeservices/1s1049/networkdrivers",
"serviceExposeMaps": "…/v1/projects/1a5/composeservices/1s1049/serviceexposemaps",
"serviceLogs": "…/v1/projects/1a5/composeservices/1s1049/servicelogs",
"storageDrivers": "…/v1/projects/1a5/composeservices/1s1049/storagedrivers",
"environment": "…/v1/projects/1a5/environment/1e148"
},
"actions": {
"remove": "…/v1/projects/1a5/composeservices/1s1049/?action=remove"
},
"baseType": "service",
"name": "code",
"state": "active",
"accountId": "1a5",
"created": "2016-12-20T04:12:46Z",
"createdTS": 1482207166000,
"currentScale": 1,
"description": null,
"externalId": null,
"fqdn": null,
"healthState": "healthy",
"instanceIds": [
"1i5032"
],
"kind": "composeService",
"launchConfig": {
"type": "launchConfig",
"externalId": "fcfd0d9c76a3a61064f51fc63e019327c2f351b804618a74991bb03cb101e835",
"hostname": "dwh",
"instanceTriggeredStop": "stop",
"kind": "container",
"nativeContainer": true,
"system": false,
"uuid": "b3ef3c96-42df-4e2b-a450-42f5a8a5ddab",
"version": "0"
},
"linkedServices": null,
"publicEndpoints": null,
"removed": null,
"scale": null,
"scalePolicy": null,
"secondaryLaunchConfigs": [ ],
"selectorContainer": "com.docker.compose.project=datawarehouse, com.docker.compose.service=code",
"selectorLink": null,
"stackId": "1st148",
"startOnCreate": true,
"system": false,
"transitioning": "no",
"transitioningMessage": null,
"transitioningProgress": null,
"uuid": "7dad4939-5c2b-44c9-a1b3-25bc196c0539",
"vip": null,
"environmentId": "1e148"
}
```
as you can see there is only one action `remove` i can perform.

Maybe this is a rancher's user permission issue, I'm trying to create a new users  and api access keys.

**Comment by bigfoot90** [comment]
I've solved my issue, thank you all for the support.

I was having two stacks `datawarehouse` and `DataWarehouse` (I don't know how this would been possible), thew script was matching the first returned by api.
Other strange thing was that this one was not been visible from the GUI (don't ask me why), I had use apis to remove it.

Once removed the wrong stack the script works like a charm now.
Thank you all for this fantastic utility.
Closing
